src/alexapi/device_platforms/orangepi.py

```diff
@@ -0,0 +1,111 @@
+import time 
+import os
+import threading
+
+from pyA20.gpio import gpio as GPIO
+
+from baseplatform import BasePlatform
+import alexapi.bcolors as bcolors
+
+
+class OrangepiPlatform(BasePlatform):
+
+	def __init__(self, config):
+		super(OrangepiPlatform, self).__init__(config)
+
+		self.__config = config
+		self.__pconfig = config['platforms']['orangepi']
+		self.__pconfig['lights'] = [self.__pconfig['rec_light'], self.__pconfig['plb_light']]
+
+		self.long_press_setup = False
+		if ('long_press' in self.__pconfig
+				and 'command' in self.__pconfig['long_press']
+				and len(self.__pconfig['long_press']['command']) > 0
+				and 'duration' in self.__pconfig['long_press']):
+
+			self.long_press_setup = True
+
+		self.should_confirm_trigger = self.__pconfig['should_confirm_trigger']
+
+		self.button_pressed = False
+
+	def setup(self):
+		GPIO.init()
+		GPIO.setcfg(self.__pconfig['button'], GPIO.INPUT)
+		GPIO.pullup(self.__pconfig['button'], GPIO.PULLUP)
+		GPIO.setcfg(self.__pconfig['rec_light'], GPIO.OUTPUT)
+		GPIO.setcfg(self.__pconfig['plb_light'], GPIO.OUTPUT)
+
+	def indicate_setup_failure(self):
+		while True:
+			for x in range(0, 5):
+				time.sleep(.1)
+				GPIO.output(self.__pconfig['rec_light'], GPIO.HIGH)
+				time.sleep(.1)
+				GPIO.output(self.__pconfig['rec_light'], GPIO.LOW)
+
+	def indicate_setup_success(self):
+		for x in range(0, 5):
+			time.sleep(.1)
+			GPIO.output(self.__pconfig['plb_light'], GPIO.HIGH)
+			time.sleep(.1)
+			GPIO.output(self.__pconfig['plb_light'], GPIO.LOW)
+
+	def after_setup(self):
+		# threaded detection of button press
+		self.wait_for_button_thread()		
+
+	def indicate_recording(self, state=True):
+		GPIO.output(self.__pconfig['rec_light'], GPIO.HIGH if state == True else GPIO.LOW)
+
+	def indicate_playback(self, state=True):
+		GPIO.output(self.__pconfig['plb_light'], GPIO.HIGH if state == True else GPIO.LOW)
+
+	def detect_button(self, channel):
```

Comment:
Just lose the `channel` parameter, it's not needed here. In RPi device file it's there because the underlying C function passes it and we have to implement it.